### PR TITLE
Feature/authorization

### PR DESCRIPTION
**Pull request by sergeyschegolev**

### CODE REVIEW

File: src/components/Header/Header.tsx [hunk 1]
```diff
@@ -21,32 +22,54 @@ const HeaderFlex = styled(Flex)`
...
-      </Flex>
-    </Box>
-  </HeaderFlex>
-);
+const Header: React.SFC<HeaderProps> = ({authStore, history}) => {
+  const signOut = () => {
+    authStore.signOut();
+  };
+
+  const singIn = () => {
```
Comment (optimusway) [inline src/components/Header/Header.tsx]: si **G** nIn

File: src/components/SignInPage/SignInPage.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+  componentDidMount() {
+    document.body.addEventListener('keyup', this.keyupHandler);
+  }
+
+  componentWillUnmount() {
+    document.body.removeEventListener('keyup', this.keyupHandler);
+  }
+
+  keyupHandler = (e: any) => {
```
Comment (optimusway) [inline src/components/SignInPage/SignInPage.tsx]: i believe handling Enter keypress can be done just by wrapping in the `form` element

File: src/components/SignInPage/SignInPage.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+  signIn = () => {
+    if (!this.state.password.length || !this.state.email.length) {
+      return;
+    }
+
+    this.props.authStore
+      .fetchBearerToken(this.state.email, this.state.password)
+      .then(() => {
+        this.props.history.push('/');
```
Comment (optimusway) [inline src/components/SignInPage/SignInPage.tsx]: not so critical, but this is not the `react-router` way

File: src/components/Header/Header.tsx [hunk 4]
```diff
@@ -21,32 +22,54 @@ const HeaderFlex = styled(Flex)`
...
+            <BalanceInfo />
+          </HeaderItem>
+          <HeaderItem>
+            <Link>
+              {authStore.isAuth ? (
+                <span className="hidden-xs" onClick={signOut}>
+                  Sign Out
+                </span>
+              ) : (
+                <span className="hidden-xs" onClick={singIn}>
```
Comment (optimusway) [inline src/components/Header/Header.tsx]: signIn

File: src/components/SignInPage/SignInPage.tsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+      })
+      .catch(error => {
+        const emailError = error.Email ? error.Email : '';
+        const passwordError = error.Password ? error.Password : '';
+        const commonError = error.message ? error.message : '';
+        this.setState({emailError});
```
Comment (optimusway) [inline src/components/SignInPage/SignInPage.tsx]: could be shorten to `this.setState({emailError, passwordError, commonError})` as i see

File: src/components/Terminal/index.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import RootStore from '../../stores/rootStore';
+import {connect} from '../connect';
+import Terminal from './Terminal';
+
+export interface TerminalProps {
+  rootStore: RootStore;
```
Comment (optimusway) [inline src/components/Terminal/index.tsx]: i might be wrong but i can't see how the `Terminal` component will get the access to `rootStore` itself. Provider injects destructured `rootStore` as before

Comment (sergeyschegolev) [reply src/components/Terminal/index.tsx]: RootStore instance is injected to the component, so we can use it's methods

File: src/components/SignInPage/SignInPage.tsx (hunk 3 above)
Comment (sergeyschegolev) [reply src/components/SignInPage/SignInPage.tsx]: Do you mean creating component using 'withRouter'?

**Pull request merged**